gerbil.evaluate/src/main/java/org/aksw/gerbil/evaluate/impl/ROCCurve.java

```diff
@@ -160,7 +162,16 @@ public double calcualteAUC() {
             pointB = points.get(i);
             // calculate area under the points (rectangle)
             if (pointB.x != pointA.x) {
-                aup = pointA.y * (pointB.x - pointA.x);
+                // if the two points are a step to the right
+                if (pointB.y == pointA.y) {
+                    aup = pointA.y * (pointB.x - pointA.x);
+                } else {
+                    // this is a diagonal
+                    // rectangle "under B"
+                    aup = pointB.y * (pointB.x - pointA.x);
```

Comment:
Doesnt it has to be `aup = pointA.y * (pointB.x - pointA.x)` 
|It would calculate the lower rectangle. 
Using `pointB.y` you'll calculate the upper rectangle. 

Maybe i am wronng here, please correct me :)